bundles/AdminBundle/Security/Authenticator/AdminAbstractAuthenticator.php

```diff
@@ -131,7 +132,7 @@ public function onAuthenticationSuccess(Request $request, TokenInterface $token,
     }
 
     /**
-     * @param $user
+     * @param User|null $user
```

Comment:
```suggestion
     * @param User $user
```
passing null has no effect so I would avoid that. Better adjust the if check.

Reply:
Do you also mean we can change it to this in Pimcore 11?
```
    protected function saveUserToSession(User $user): void
    {
        if (Authentication::isValidUser($user->getUser())) {
```

The parameter types should not be what has effect, but what types may be passed.

Reply:
yes. this should require `User` type to be passed on Pimcore 11.

Reply:
I changed it: https://github.com/pimcore/pimcore/pull/12862/commits/ddcad9d2c3b53614b169de3d18900c948285172a